File: app/src/main/java/pl/krystiankaniowski/futuremind/model/database/ModelConverter.java
package pl.krystiankaniowski.futuremind.model.database;

import java.util.regex.Pattern;

import pl.krystiankaniowski.futuremind.model.rest.SingleData;

public class ModelConverter {

    private static final Pattern PATTERN = Pattern.compile("http://.*");

    public static Row toDatabaseModel(SingleData data) {

        Row row = new Row();

        row.setTitle(data.getTitle());
        row.setDescription(data.getDescription());
        row.setOrderId(data.getOrderId());
        row.setImageUrl(data.getImageUrl());
        row.setUrl(data.getDescription());
        row.setModificationDate(data.getModificationDate());

        return row;

    }

}

File: app/src/main/java/pl/krystiankaniowski/futuremind/model/ModelConverter.kt
package pl.krystiankaniowski.futuremind.model

import pl.krystiankaniowski.futuremind.model.database.Row
import pl.krystiankaniowski.futuremind.model.rest.SingleData
import java.util.*

class ModelConverter {

    companion object {

        val TAG: String? = ModelConverter.javaClass.simpleName

        fun toDatabaseModel(data: SingleData): Row {

            val addressStartPostion = data.description!!.indexOf("http://")

            val row = Row()

            row.title = data.title
            row.description = data.description!!.substring(0, addressStartPostion).trim()
            row.orderId = data.orderId!!
            row.imageUrl = data.imageUrl
            row.url = data.description!!.substring(addressStartPostion)
            row.modificationDate = data.modificationDate

            return row

        }

        fun toDatabaseModel(data: List<SingleData>): List<Row> {

            var list = ArrayList<Row>()

            for (item in data) {
                list.add(toDatabaseModel(item))
            }

            return list

        }

    }

}
